src/artifacts.js

```diff
@@ -63,40 +63,23 @@ export default class Artifacts {
   }
 
   /**
-  Find and extract an artifact if possible returns true if one is
-  found/extracted false otherwise...
+  Find a possible cached archive and download if possible.
   */
-  async useIfAvailable(name, namespace, dest) {
+  async downloadIfAvailable(name, namespace, dest) {
     let localPath = this.lookupLocal(name);
+
     if (await fs.exists(localPath)) {
-      // Attempt to extract from local tar if this fails then allow remote
-      // download to be attempted...
-      try {
-        await this.extract(localPath, dest);
-        return true;
-      } catch (e) {
-        console.error(
-          'Error extracting tar re-downloading',
-          localPath,
-          e.stack
-        );
-        // Destroy cache...
-        await run(`rm -Rf ${localPath} ${dest}`);
-        // Intentionally falling through...
-      }
+      return localPath;
```

Comment:
So far out of the hundreds of runs I've seen, I haven't seen this issue hit.  I'm wondering if this happened for A) corrupted downloaded which we should be solving in other ways B) this was actually because things were attempting to be extracted in parallel.